### ISSUE TITLE
[Favorite] Favorite file moved into a shared folder as recipient is no longer favorite

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/core#13274]
### Steps to reproduce

1. Make a file or folder favorite
2. Move it inside a shared folder by another user.

### Expected behaviour
File is still favorite.

### Actual behaviour
File loses its status as favorite file.

### Server configuration
**Operating system**:
Ubuntu 14.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.0.0.1","versionstring":"8.0 alpha 1","edition":""}

**Updated from an older ownCloud or fresh install:**
fresh

**List of activated apps:**
none

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password and passwordsalt)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes

### Client configuration
**Browser:**
Chrome

**Comment by nickvergessen** [comment]
Also happens when moving a shared favorited file out of the share.
I guess this is due to the fact that the file id changes?

@PVince81

**Comment by DeepDiver1975** [comment]
@PVince81 can you please have a look? THX

**Comment by nickvergessen** [comment]
Also happens with external storages

**Comment by PVince81** [comment]
I'll have a look. It is likely due to file id change.
I suspect we'll need to spend some time in the future to make file ids stable...

**Comment by PVince81** [comment]
Just checked the database and indeed the file id changes.
It's not only causing the favorite info to be lost, but also the share info!
If you share the file with link, then move it into a shared folder. Boom. Share lost. I'll make a separate ticket.

**Comment by PVince81** [comment]
Raised the share breakage here: https://github.com/owncloud/core/issues/13307

One possible solution is to add additional hooks that checks for such situations and then automatically update the share/favorite mapping by updating the file id. But honestly I'd prefer if we had stable file ids instead. There's more and more stuff relying on file ids.

**Comment by PVince81** [comment]
Raised https://github.com/owncloud/core/issues/13310 to make the file id stable in general.

**Comment by PVince81** [comment]
File id stability https://github.com/owncloud/core/issues/13310 is not fixable in 8.0. Deferring to 8.1 as per the blocking ticket.

**Comment by MorrisJobke** [comment]
Fixed with #15360
